dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -1747,22 +1747,22 @@ DomainParticipantImpl::create_new_topic(
   return DDS::Topic::_duplicate(refCounted_topic.pair_.obj_.in());
 }
 
-int
+bool
 DomainParticipantImpl::is_clean() const
 {
-  int sub_is_clean = subscribers_.empty();
-  int topics_is_clean = topics_.size() == 0;
+  bool sub_is_clean = subscribers_.empty();
+  bool topics_is_clean = topics_.size() == 0;
 
   if (!TheTransientKludge->is_enabled()) {
     // There are four topics and builtin topic subscribers
     // left.
 
-    sub_is_clean = sub_is_clean == 0 ? subscribers_.size() == 1 : 1;
-    topics_is_clean = topics_is_clean == 0 ? topics_.size() == 4 : 1;
+    sub_is_clean = !sub_is_clean ? subscribers_.size() == 1 : true;
```

Comment:
Not part of your change, but shouldn't this be checking if Built-In Topics are enabled before assuming one subscriber and four topics?